applications/Preprocessor/lib/output_impl.h

```diff
@@ -102,11 +103,18 @@ void outputMesh(Mesh<dimension>& mesh,
     }
     std::ofstream output(outputFileName.getValue());
     output << std::hexfloat;
-    output << "mesh 1" << std::endl;
+    size_t version = 2;
+    output << "mesh " << version << std::endl;
     output << mesh.getNumberOfNodes() << " " << mesh.getNumberOfElements()
            << " " << dimension;
     printOtherEntityCounts(output, mesh, Detail::tag<dimension - 1>{});
     output << std::endl;
+    output << "zones" << std::endl;
```

Comment:
I think the number of zones should be included here. Something like the following:

```suggestion
    output << "zones" << std::endl;
    output << mesh.getZoneNames().size() << std::endl;
```